### DISCUSSION TITLE
Memory leak when we have nested for loop

### DISCUSSION THREAD

**Discussion by RavalikaP** [nodejs/help#5052] [Q&A]
### Version

12.6

### What steps will reproduce the bug?

```js
attrs = ['cat1','cat2','cat3','cat4','cat5','cat6','cat7','cat8'];
map = {cat1: 1,cat2: 2,cat3: 3,cat4: 4,cat5: 5,cat6: 6,cat7: 7,cat8: 8 };

 for (let recordIndex = 0; recordIndex < 500000; recordIndex++) {
      record = data[recordIndex];
      uiRecord = {};
      for (let attrIndex = 0; attrIndex < 8; attrIndex++) {
        //let attr = attrs[attrIndex];
        uiRecord[attrs[attrIndex]] = record[map[attrs[attrIndex]]];
      }
}
```

### How often does it reproduce? Is there a required condition?

Memory is increasing 750mb for this present loop.

### What is the expected behavior?

I want to know the reason why it is going to 750mb

### What do you see instead?

Want to know why it is having so much memory leak.

**Comment by RavalikaP** [comment]
Can anyone respond to this?

**Comment by aduh95** [comment]
> Memory is increasing 750mb for this present loop.

What do you mean by that? How do you measure that?

> Want to know why it is having so much memory leak.

What makes you think there's a memory leak? To me it looks like you are creating a huge number of JS objects, and that takes space in the heap. In order to remain efficient, V8 won't run garbage collector in your loop if your machine has enough memory at its disposal; a memory leak would be if the garbage collector doesn't free up the memory, but you haven't given enough info to draw this conclusion.
Note that the code snippet you provided is using global variables, those won't be freed up by the garbage collector (but given the size of the final objects, that should not be a significant amount).
